types/process.d.ts

```diff
@@ -14,5 +18,305 @@ declare module "process" {
         | "SIGSEGV"
         | "SIGTERM";
     }
+
+    type Platform =
+      | "aix"
+      | "android"
+      | "darwin"
+      | "freebsd"
+      | "haiku"
+      | "linux"
+      | "openbsd"
+      | "sunos"
+      | "win32"
+      | "cygwin"
+      | "netbsd";
+
+    type Architecture =
+      | "arm"
+      | "arm64"
+      | "ia32"
+      | "loong64"
+      | "mips"
+      | "mipsel"
+      | "ppc"
+      | "ppc64"
+      | "riscv64"
+      | "s390"
+      | "s390x"
+      | "x64";
+  }
```

Comment:
Lets be diligent about typing, we dont support nearly half of those platforms and architecture.